log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/PatternParser.java

```diff
@@ -669,20 +668,20 @@ private int finalizeConverter(final char c, final String pattern, final int star
                 msg.append("] starting at position ");
             }
 
-            msg.append(Integer.toString(i));
+            msg.append(i);
             msg.append(" in conversion pattern.");
```

Comment:
I do not think we should hide boxing and unboxing if we are making all of these performance based changes. I would think we would want to see as much of the mechanics as possible.

Reply:
This actually uses the `append(int)` overload, avoiding boxing and string allocation entirely :-)

https://github.com/openjdk/jdk/blob/b4b121018d16e531f3a51ff75ae37fdc374d530b/src/java.base/share/classes/java/lang/AbstractStringBuilder.java#L811-L834